Sources/NIOKit/FutureExtensions.swift

```diff
@@ -0,0 +1,13 @@
+// MARK: - Guard
+
+public extension EventLoopFuture {
+	/// Guards that the future's value satisfies the callback's condition or fails with the given error.
+	func `guard`(_ callback: @escaping ((Value) -> Bool), else error: Error) -> EventLoopFuture<Value> {
+		let promise = eventLoop.makePromise(of: Value.self)
+		whenSuccess {
+			if callback($0) { promise.succeed($0) } else { promise.fail(error) }
+		}
+		whenFailure { promise.fail($0) }
```

Comment:
i think using `whenComplete` and switching on the result would be more performant than adding two closures to the future.